lib/Sema/CSSimplify.cpp

```diff
@@ -6951,6 +6963,32 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
     return matchTypes(type1, type2, matchKind, subflags, locator);
   }
 
+  case FixKind::AllowTupleTypeMismatch: {
+    auto lhs = dyn_cast<TupleType>(type1.getPointer());
+    auto rhs = dyn_cast<TupleType>(type2.getPointer());
+    auto lhsLarger = lhs->getElements().size() >= rhs->getElements().size();
+    auto larger = lhsLarger ? lhs : rhs;
+    auto smaller = lhsLarger ? rhs : lhs;
+    if (lhs && rhs && getContextualTypePurpose() == CTP_Initialization) {
```

Comment:
This condition should probably be a precondition for creating of `AllowTupleTypeMismatch` as I mentioned in my previous comment.